### PR TITLE
Add Kubernetes Distro and ICP Prereqs to specification

### PR DESCRIPTION
**Pull request by cdjohnson**

### CODE REVIEW

**Comment by cdjohnson** [comment]
Note:  There is a reference to the ICP 3.2 doc that is not yet available.  This is by design.

File: samples/utilities/checkPrereqs.sh [hunk 1]
```diff
@@ -0,0 +1,582 @@
+#/bin/sh
+
```
Comment (morstad) [inline samples/utilities/checkPrereqs.sh]: Do we need some type of license - apache or ?

File: samples/utilities/checkPrereqs.sh [hunk 2]
```diff
@@ -0,0 +1,582 @@
...
+ICP_CLUSTER_HEALTH_URL=
+setup_endpoints() {
+  ICP_MGMT_ENDPOINT="$(get_icp_endpoint)"
+  ICP_CLUSTER_HEALTH_URL="$ICP_MGMT_ENDPOINT/cluster-health/v1alpha1/clusterstatus"
+}
+
+# The cluster-health service uses the Kube token.
+AUTHZTOKEN=
+setup_authz_token() {
+  # "$(cloudctl tokens --id)"
```
Comment (morstad) [inline samples/utilities/checkPrereqs.sh]: remove obsolete reference to cloudctl

File: samples/utilities/checkPrereqs.sh [hunk 3]
```diff
@@ -0,0 +1,582 @@
...
+}
+
+strindex() { 
+  x="${1%%$2*}"
+  [[ "$x" = "$1" ]] && echo -1 || echo "${#x}"
+}
+
+# stdout:  The IBM Cloud Private endpoint URL
+get_icp_endpoint() {
+  # $ cloudctl api
```
Comment (morstad) [inline samples/utilities/checkPrereqs.sh]: remove unused command

File: samples/utilities/checkPrereqs.sh [hunk 4]
```diff
@@ -0,0 +1,582 @@
...
+#
+# Dump all the service names known by this tool.
+#  checkPrereqs.sh list
+# Dump all the installed services for the connected cluster
+#  checkPrereqs.sh check --allicpservices
+# Test for selected services to see if they are installed.
+#  checkPrereqs.sh check --icpservices audit-logging,cert-manager
+# 
+# A non-zero exit code indicates that at least one of the specified services is NOT installed.
+
```
Comment (morstad) [inline samples/utilities/checkPrereqs.sh]: With next release please consider setting recommended error handling : 

# Set error handling
set -o errexit
set -o nounset
set -o pipefail

File: samples/utilities/checkPrereqs.sh (hunk 1 above)
Comment (cdjohnson) [reply samples/utilities/checkPrereqs.sh]: License in repo governs this script.

**Pull request merged**